pkg/apis/autoscaling/v1alpha1/hvpa_types.go

```diff
@@ -27,23 +27,52 @@ import (
 
 // VpaTemplateSpec defines the spec for VPA
 type VpaTemplateSpec struct {
+	// Describes the rules on how changes are applied.
+	// If not specified, all fields in the `UpdatePolicy` are set to their
+	// default values.
+	// +optional
+	UpdatePolicy *UpdatePolicy `json:"updatePolicy,omitempty" protobuf:"bytes,1,opt,name=updatePolicy"`
 	// Controls how the autoscaler computes recommended resources.
 	// The resource policy may be used to set constraints on the recommendations
 	// for individual containers. If not specified, the autoscaler computes recommended
 	// resources for all containers in the pod, without additional constraints.
 	// +optional
-	ResourcePolicy *vpa_api.PodResourcePolicy `json:"resourcePolicy,omitempty" protobuf:"bytes,3,opt,name=resourcePolicy"`
+	ResourcePolicy *vpa_api.PodResourcePolicy `json:"resourcePolicy,omitempty" protobuf:"bytes,2,opt,name=resourcePolicy"`
+}
+
+// UpdatePolicy describes the rules on how changes are applied.
+type UpdatePolicy struct {
+	// Controls when autoscaler applies changes to the resources.
+	// The default is 'On'.
+	// +optional
+	UpdateMode *UpdateMode `json:"updateMode,omitempty" protobuf:"bytes,1,opt,name=updateMode"`
 }
 
+// UpdateMode controls when autoscaler applies changes to the resoures.
+type UpdateMode string
```

Comment:
Could it make sense to have different `UpdateMode` types for VPA and HPA? Then we can have stricter enumeration/validation. I.e. `VPAUpdateMode` supports only `Off`, `Auto` and `ScaleUp` while `HPAUpdateMode` supports only `Off`, `Auto` and `ScaleOut`. WDYT?

Reply:
Yes, makes sense to keep it separate. Will make the changes.
Thanks for the suggestion. :)